C/errorCodes.h

```diff
@@ -15,7 +16,23 @@
 #define SIMPLICITY_ERR_FAIL_CODE (-8)
 #define SIMPLICITY_ERR_STOP_CODE (-10)
 #define SIMPLICITY_ERR_HIDDEN (-12)
+#define SIMPLICITY_ERR_BITSTREAM_UNUSED_BYTES (-14)
+#define SIMPLICITY_ERR_BITSTREAM_UNUSED_BITS (-16)
+#define SIMPLICITY_ERR_TYPE_INFERENCE_UNIFICATION (-18)
+#define SIMPLICITY_ERR_TYPE_INFERENCE_OCCURS_CHECK (-20)
+#define SIMPLICITY_ERR_TYPE_INFERENCE_NOT_PROGRAM (-22)
+#define SIMPLICITY_ERR_WITNESS_EOF (-24)
+#define SIMPLICITY_ERR_WITNESS_UNUSED_BITS (-26)
+#define SIMPLICITY_ERR_UNSHARED_SUBEXPRESSION (-28)
+#define SIMPLICITY_ERR_CMR (-30)
+#define SIMPLICITY_ERR_AMR (-32)
+#define SIMPLICITY_ERR_EXEC_BUDGET (-34)
+#define SIMPLICITY_ERR_EXEC_MEMORY (-36)
+#define SIMPLICITY_ERR_EXEC_JET (-38)
+#define SIMPLICITY_ERR_EXEC_ASSERT (-40)
+#define SIMPLICITY_ERR_ANTIDOS (-42)
 
-#define PERMANENT_FAILURE(err) (!((err) & 1))
+/* Check if failure is permanent or success (which is always permanent). */
+#define PERMANENT(err) (!((err) & 1))
```

Comment:
In 379a00037f08348746dc9dd3940600f4f798e610:

I'd prefer if this were called `IS_PERMANENT` to be clear that it's checking a condition rather than marking an existing error as permanent somehow.